src/nvim/search.c

```diff
@@ -4417,8 +4417,8 @@ find_pattern_in_path (
         i = (int)(p - aux);
 
         if ((compl_cont_status & CONT_ADDING) && i == compl_length) {
-          /* IOSIZE > compl_length, so the STRNCPY works */
-          STRNCPY(IObuff, aux, i);
+          // IOSIZE > compl_length, so the memcpy works
+          memcpy(IObuff, aux, i);
```

Comment:
That was an oversight, thanks. The other one can be memcpy, it's nul-terminated immediately.